### PR TITLE
MM-52641 MM-52645 Playwright/Accessibility: Initial setup and accessibility fix on login page

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
- Added `@axe-core/playwright` for Accessibility testing with Playwright
- Put in a separate folder `e2e-tests/playwright/tests/accessibility` since this will run separately and with focus on accessibility testing only. Folder structure in this path would be:
  - `.../tests/accessibility/[common|boards|channels|playbooks]` 
  - note that Calls will have their own accessibility test in its own repo

Steps on how to run accessibility test using FastPass - https://accessibilityinsights.io/docs/web/getstarted/fastpass :
1. Download and install FastPass as browser extension - https://accessibilityinsights.io/downloads/.
2. Visit login page of a Mattermost instance
3. Run automated check and see failed accessibility instances - https://accessibilityinsights.io/docs/web/getstarted/fastpass/#run-automated-checks 

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-52641
- https://mattermost.atlassian.net/browse/MM-52645

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@saturninoabril thank you for setting up a11y support in tests

File: webapp/channels/src/root.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <head>
     <meta charset="utf-8">
 
-    <meta name='viewport' content='width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0'>
+    <meta name='viewport' content='width=device-width, initial-scale=1.0'>
```
Comment (saturninoabril) [inline webapp/channels/src/root.html:6]: This fixes the issue of `Ensures <meta name="viewport"> does not disable text scaling and zooming.` See https://accessibilityinsights.io/info-examples/web/meta-viewport/

File: webapp/channels/src/components/header_footer_route/header.tsx [hunk 2]
```diff
@@ -21,12 +21,15 @@ export type HeaderProps = {
...
 
+    const ariaLabel = EnableCustomBrand === 'true' && SiteName ? SiteName : 'Mattermost';
+
     return (
         <div className='hfroute-header'>
             <div className='header-main'>
                 <Link
                     className='header-logo-link'
                     to='/'
+                    aria-label={ariaLabel}
```
Comment (saturninoabril) [inline webapp/channels/src/components/header_footer_route/header.tsx:32]: This fixes the issue of `Ensures buttons have discernible text `. See https://accessibilityinsights.io/info-examples/web/link-name/

File: webapp/channels/src/components/widgets/inputs/password_input/password_input.tsx [hunk 3]
```diff
@@ -73,11 +78,7 @@ const PasswordInput = React.forwardRef((
             onChange={onChange}
             onBlur={onBlur}
             onFocus={onFocus}
-            placeholder={createMode ? (
-                formatMessage({id: 'widget.passwordInput.createPassword', defaultMessage: 'Choose a Password'})
-            ) : (
-                formatMessage({id: 'widget.passwordInput.password', defaultMessage: 'Password'})
-            )}
+            placeholder={placeHolder}
```
Comment (saturninoabril) [inline webapp/channels/src/components/widgets/inputs/password_input/password_input.tsx:81]: These password and login input fix the issue of `Ensures buttons have discernible text `. See https://accessibilityinsights.io/info-examples/web/button-name/

**Comment by saturninoabril** [comment]
Hello @amyblais, I'd suggest to get this into v8.1. Thanks.

**Review by yasserfaraazkhan — APPROVED** [review]

File: e2e-tests/playwright/tests/functional/channels/drafts/drafts_on_deleted_message.spec.ts [hunk 4]
```diff
@@ -52,7 +52,7 @@ test('MM-T5435_1 Global Drafts link in sidebar should be hidden when another use
     await sidebarRight.postMessage(draftMessageByUser);
 
     // # Close the RHS for draft to be saved
-    await sidebarRight.close()
+    await sidebarRight.close();
```
Comment (M-ZubairAhmed) [inline e2e-tests/playwright/tests/functional/channels/drafts/drafts_on_deleted_message.spec.ts:55]: hmm i see we have currently no way to catch these 🤔

File: webapp/channels/src/root.html [hunk 5]
```diff
@@ -3,7 +3,7 @@
 <head>
     <meta charset="utf-8">
 
-    <meta name='viewport' content='width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0'>
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/root.html:6]: just wondering why did we had `user-scalable=0` before

File: webapp/channels/src/root.html (hunk 1 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/root.html:6]: thank you for removing this. user-scalable=0 is not a recommended approach

File: e2e-tests/playwright/tests/accessibility/common/login.spec.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+test('/login', async ({pw, pages, page, axe}) => {
+    // # Go to login page
+    const {adminClient} = await pw.getAdminClient();
+    const adminConfig = await adminClient.getConfig();
+    const loginPage = new pages.LoginPage(page, adminConfig);
+    await loginPage.goto();
+    await loginPage.toBeVisible();
+
+    // # Analyze the page
+    const accessibilityScanResults = await axe.loginPage().analyze();
```
Comment (M-ZubairAhmed) [inline e2e-tests/playwright/tests/accessibility/common/login.spec.ts:15]: so takes into account all a11y issues consideration as per axe. Awesome

**Review by M-ZubairAhmed — APPROVED** [review]

File: e2e-tests/playwright/tests/functional/channels/drafts/drafts_on_deleted_message.spec.ts (hunk 4 above)
Comment (saturninoabril) [reply e2e-tests/playwright/tests/functional/channels/drafts/drafts_on_deleted_message.spec.ts:55]: I see. Will address separately.

**Review by mvitale1989 — APPROVED** [review]
Thanks Saturn, LGTM!

**Pull request merged**